src/library/Cardano/CoinSelection/Fee.hs

```diff
@@ -98,20 +98,41 @@ newtype Fee = Fee { unFee :: Coin }
     deriving stock (Eq, Generic, Ord)
     deriving Show via (Quiet Fee)
 
--- | Defines the maximum size of a dust coin.
+-- | Defines the /maximum/ size of a __dust coin__.
 --
--- Change values that are less than or equal to this threshold will not be
--- included in coin selections produced by the 'adjustForFee' function.
+-- Functions that accept a 'DustThreshold' argument will generally exclude
+-- values that are /less than or equal to/ this threshold from the 'change'
+-- sets of generated selections, /coalescing/ such coins together into larger
+-- coins that /exceed/ the threshold.
+--
+-- Specifying a dust threshold of __/n/__ causes all coins that are less than
+-- or equal to __/n/__ to be treated as dust and coalesced together.
+--
+-- Specifying a dust threshold of __0__ completely /disables/ dust elimination
+-- with the exception of zero-valued coins, which will always be eliminated.
+--
+-- See 'coalesceDust'.
 --
 newtype DustThreshold = DustThreshold { unDustThreshold :: Coin }
     deriving stock (Eq, Generic, Ord)
     deriving Show via (Quiet DustThreshold)
 
--- | Provides a function capable of estimating the fee for a given coin
---   selection.
+-- | Provides a function capable of __estimating__ the transaction fee required
+--   for a given coin selection, according to the rules of a particular
+--   blockchain.
+--
+-- The fee estimate should be a function of the __current__ memberships of the
+-- 'inputs', 'outputs', and 'change' sets.
 --
--- The fee estimate can depend on the numbers of inputs, outputs, and change
--- outputs within the coin selection, as well as their magnitudes.
+-- Depending on the rules of the blockchain under consideration, the fee
+-- estimate may take either (or both) of the following factors into account:
+--
+--   - the number of entries in each set;
+--   - the coin value of each set member.
```

Comment:
when I was thinking about it one day I was also thinking that `relative coin values could also be taken into account` or `distribution of coin values` ...

Reply:
> when I was thinking about it one day I was also thinking that `relative coin values could also be taken into account` or `distribution of coin values` ...

These could also be factors! (Since in principle, this library could be used with any blockchain.)